tools/mo/openvino/tools/mo/moc_frontend/pipeline.py

```diff
@@ -96,9 +96,12 @@ def add_names_to_tensors(model: InputModel, places: List[Place]):
         new_input_places = [x['node'] for x in user_shapes]
         input_model.override_all_inputs(new_input_places)
         # invalidation of existing Place objects could have happened in the operation above
+        names = [place.get_names()[0] for place in new_input_places]
+        shapes = [shape for shape in argv.placeholder_shapes.values()]
+        placeholder_shapes = dict(zip(names, shapes))
```

Comment:
We have to update the names, because on the second iteration of `decode_name_with_port` called in pipeline.py, after cutting the input, when inputs are not equal, there is no longer node with the name `node_name`. As a basic example in the mnist-8.onnx model we have the node with name `ReLU114`. If we use `mo.py` for this model with the flag `
--input=ReLU114` in the first iteration of calling decode_with_port_name (before cutting the graph) the node with name `ReLU114` is found, but in the second iteration when it becomes the input of the graph its name is changed to `ReLU114_Output_0` (its output tensor), so if we do not update the node name in this line, we will get the error "[ ERROR ]  No node with name ReLU114" which was the source of ticket for this PR